pal/src/map/virtual.cpp

```diff
@@ -878,15 +870,14 @@ static LPVOID VIRTUALReserveMemory(
     MemSize = ( ((UINT_PTR)lpAddress + dwSize + VIRTUAL_PAGE_MASK) & ~VIRTUAL_PAGE_MASK ) -
                StartBoundary;
 
-    InternalEnterCriticalSection(pthrCurrent, &virtual_critsec);
-
-    // If this is a request for special executable (JIT'ed) memory then, first of all,
-    // try to get memory from the executable memory allocator to satisfy the request.
-    if (((flAllocationType & MEM_RESERVE_EXECUTABLE) != 0) && (lpAddress == NULL))
+    if ((flAllocationType & MEM_RESERVE_EXECUTABLE) != 0)
     {
-        pRetVal = g_executableMemoryAllocator.AllocateMemory(MemSize);
+        fprintf(stderr, "MEM_RESERVE_EXECUTABLE is not supported!");
```

Comment:
Actually, we do have a preReservedSegment in the JIT (for CFG cases)- it's turned off on xplat because CFG support isn't there but I think clang's CFI (https://clang.llvm.org/docs/ControlFlowIntegrity.html) provides the same support so we should look to turning it back on soon. In which case, if the preReservedSegment mimicked this mechanism, I think it could be nice. @curtisman thoughts?